modules/install/wizard.php

```diff
@@ -199,14 +199,72 @@
 
             if ($res == 1 or $res == 3 or $res == 5) {
                 $db->connect();
+                
+            if ($db->success) {
+                
+                $dsp->AddFieldSetStart(t('MySQL'));
+
+                // SQL mode - check for options with known issues
+                $sqlmode_disable = array('ONLY_FULL_GROUP_BY');
+                $res = $db->qry('SELECT @@SESSION.SQL_MODE as sqlmode;');
+                $opts = $db->fetch_array($res)['sqlmode'];
+                $server_opts = explode (',',$opts);
+                $warn_opts = implode(array_values(array_intersect($server_opts,$sqlmode_disable)),',');
+                $sql_mode='';
+                if ($warn_opts) {
+                    $sql_mode = $warning . t("Der MySQL-Server hat aktuell folgende Optionen aktiviert, die zu Problemen mit LanSuite führen: %1 \nEs wird versucht, dies automatisch zu kompensieren, indem diese für jede Verbindung deaktiviert werden. Dies hat zur Folge, dass eventuelle Änderungen an der Server-Variable SQL_MODE nicht automatisch für LanSuite übernommen werden, sondern manuell in config.php angepasst werden müssen", $warn_opts);
```

Comment:
If I remember correctly, then this is most probably a copy&paste mistake from [modules](https://github.com/lansuite/lansuite/tree/master/modules)/[install](https://github.com/lansuite/lansuite/tree/master/modules/install)/[Classes](https://github.com/lansuite/lansuite/tree/master/modules/install/Classes)/[Install.php](https://github.com/lansuite/lansuite/tree/master/modules/install/Classes/Install.php)

There is a block of style definitions that would also be needed here:

```
$ok = "<span class=\"okay\">".t('Erfolgreich')."</span>" . HTML_NEWLINE;
$failed = "<span class=\"error\">".t('Fehlgeschlagen')."</span>" . HTML_NEWLINE;
$warning = "<span class=\"warning\">".t('Bedenkliche Einstellung')."</span>" . HTML_NEWLINE;
$optimize = "<span class=\"warning\">".t('Optimierungsbedarf')."</span>" . HTML_NEWLINE;
$not_possible = "<span class=\"warning\">".t('Leider nicht möglich')."</span>" . HTML_NEWLINE;
```
There is also a formatting variable here that can be used, but that is for reasons also only in one case of a prior conditional and thus would need to be moved out to be usable:
```
        $fail_leadin = "<font color=\"#ff0000\">";
        $leadout = "</font>";
```

I think this was noted as code duplication by sonarcloud and I wanted to move that to a separate function somewhere....I think?

Reply:
I know what you mean. In my opinion, this is because the reporting is implemented in the wrong section (at Step 3). See https://github.com/lansuite/lansuite/pull/476#pullrequestreview-1448384547